src/components/network-map-tab.js

```diff
@@ -194,78 +183,123 @@ export const NetworkMapTab = ({
         []
     );
 
+    const getMissingEquipmentsPositions = useCallback(
+        (foundEquipmentPositions, allEquipments, fetchEquipmentCB) => {
+            let notFoundEquipmentsIds = [];
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+            allEquipments.forEach((s) => {
+                if (!foundEquipmentsIds.includes(s.id)) {
+                    notFoundEquipmentsIds.push(s.id);
+                }
+            });
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+            setWaitingLoadGeoData(true);
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
     const reloadMapGeoData = useCallback(() => {
-        console.info(`Loading geo data of study '${studyUuid}'...`);
-        const substationPositions = fetchSubstationPositions(
-            studyUuid,
-            currentNode?.id
-        );
-        const linePositions = lineFullPath
-            ? fetchLinePositions(studyUuid, currentNode?.id)
-            : [];
-        setWaitingLoadGeoData(true);
-
-        Promise.all([substationPositions, linePositions])
-            .then((values) => {
-                const newGeoData = new GeoData();
-                newGeoData.setSubstationPositions(values[0]);
-                newGeoData.setLinePositions(values[1]);
-                setGeoData(newGeoData);
-                setWaitingLoadGeoData(false);
-            })
-            .catch(function (error) {
-                console.error(error.message);
-                setWaitingLoadGeoData(false);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
+        if (studyUuid && currentNode) {
+            if (
+                geoData &&
+                (geoData.substationPositionsById.size > 0 ||
+                    geoData.linePositionsById.size > 0)
+            ) {
+                const missingSubstationPositions =
```

Comment:
add a console info ? as for the global geodata fetch :
console.info(`Loading geo data of study '${studyUuid}' for missing equipments...`);

Reply:
It's a bit redondant as we already have this log in the fetch : 
_Fetching substations positions of study 'bbd4f4e1-a0f3-491c-a5e9-a9b5a2d2e82f' and node '5496ec90-4f5f-4cc6-a3b6-bf91f4fb50d6' with ids '_
And in most of case, there will be no fetch as all required geodata will be already loaded